modules/programs/i3lock.nix

```diff
@@ -0,0 +1,291 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.i3lock;
+
+  enablei3LockColor = any (value: value != null || value != false) [
+    cfg.blur
+    cfg.clock.enable
+    cfg.verifyText
+    cfg.wrongText
+    cfg.textSize
+    cfg.modSize
+    cfg.colors
+  ];
+
+  bin = if enablei3LockColor then "i3lock-color" else "i3lock";
+
+  lockCmd = concatStringsSep " " (filter (value: value != "") [
+    "${cfg.package}/bin/${bin}"
+
+    # general
+    (optionalString (cfg.color != null) "--color=${cfg.color}")
+    (optionalString cfg.beep "--beep")
+    (optionalString (cfg.pointer != null) "--pointer=${cfg.pointer}")
+    (optionalString (cfg.image != null) "--image=${cfg.image}")
+    (optionalString (cfg.blur != null) "--blur=${toString cfg.blur}")
+    (optionalString enablei3LockColor "--indicator")
+
+    # clock
+    (optionalString cfg.clock.enable "--clock")
+    (optionalString cfg.clock.enable "--timestr=${cfg.clock.timeStr}")
+    (optionalString cfg.clock.enable "--datestr=${cfg.clock.dateStr}")
+    (optionalString (cfg.clock.enable && (cfg.clock.timeFont != null))
+      "--timefont=${cfg.clock.timeFont}")
+    (optionalString (cfg.clock.enable && (cfg.clock.dateFont != null))
+      "--datefont=${cfg.clock.dateFont}")
+
+    # text
+    (optionalString (cfg.textSize != null) "--datefont=${cfg.textSize}")
```

Comment:
You probably wanted to use this options from i3lock-color(1):

> --{time, date, layout, verif, wrong}size=number
              Sets the font size used to render various strings.

Reply:
Implemented said options